plutus-benchmark/marlowe/src/PlutusBenchmark/Marlowe/Core/V1/Semantics/Types.hs

```diff
@@ -97,12 +67,20 @@ deriving stock instance Data Ledger.ScriptHash
 deriving stock instance Data Ledger.StakingCredential
 
 
--- | A Party to a contract.
-data Party =
-    Address Network Ledger.Address  -- ^ Party identified by a network address.
-  | Role TokenName                  -- ^ Party identified by a role token name.
-  deriving stock (Generic,Haskell.Eq,Haskell.Ord,Haskell.Show,Data)
+$(asData
```

Comment:
Hmm this is not super nice syntax, and may not work very well with HLS. How about the following?

```haskell
-- | A Party to a contract.
data _Party =
    _Address Network Ledger.Address  -- ^ Party identified by a network address.
  | _Role TokenName                  -- ^ Party identified by a role token name.

-- Generates newtype Party and pattern synonyms
makeData ''_Party
```

Reply:
HLS won't have any more trouble with the `$()` TH quotes than the top level ones - the only difference with the latter is that the parser inserts the implicit `$()` around `makeData ''_Party`. 

And it's not like having the separate data declaration helps you otherwise: it just adds extra names that you don't want.

Reply:
Actually, this is also a top-level splice, so you don't need the `$( )` either. It's still not lovely, but that makes it a bit better. I'm not going to go through this and change it though, you can see it in https://github.com/input-output-hk/plutus/pull/5512